include/compartment.h

```diff
@@ -55,6 +55,9 @@ struct intercept_patch
     void* __capability manager_cap;
 };
 
+// Maximum size of an argument, in bytes
+#define COMP_ARG_SIZE 8
```

Comment:
Is this because we use `ptrsize_t` (or similar type) wide arguments? If so, should we do `sizeof(<appropriate type>)` or are things hard coded to `uint64_t`?

Reply:
I think it would be worth thinking about whether this can be tied to a C type. If it can't, then ideally `#ifdef` it so that if someone compiles it on a different (e.g. 32-bit) platform things don't go splat :)

Reply:
I think it will go splat at compilation, because (as far as I know), there are no `x` registers in 32-bit land, so `transition.S` would need to be manually changed to use `w` registers. And I think there are more issues with compiling for 32-bits than just this (e.g., I'm not sure if `sizeof(long long)` is defined, or how [1]).

I can try compiling for a 32-bit platform and see what happens.

[1] https://github.com/capablevms/CHERI-ELF-comp/pull/7/files#diff-adfde91070db01742cb89ede1d5f00ff6ca15a8f01b6606c1bae07ee95b132ddR299-R300

Reply:
We don't need to test it on another platform: but I've learnt over time that if we hardcode an architecture specific magic number in to guard it to avoid future confusion.

Reply:
I couldn't find any C data type that corresponds to the size of a register. I think the best solution is to add a `static_assert(sizeof(intptr_t) == COMP_ARG_SIZE)` under this definition (with an appropriate comment).

Reply:
Added in https://github.com/capablevms/CHERI-ELF-comp/pull/7/commits/dc96f5576d9a1aeb94fc1936b3ac714001736f09.

Reply:
I have updated the guard similar to what we discussed earlier today in https://github.com/capablevms/CHERI-ELF-comp/pull/7/commits/092df891904dec4e36226d44d5e62bd54fe8124f. I was unable to find a `MORELLO_HYBRID` macro, so I've left a todo in case we'll find one at some time (or we can refine this check).

Reply:
OK.